src/Compilers/Core/Portable/InternalUtilities/FatalError.cs

```diff
@@ -225,7 +225,7 @@ public static bool ReportAndCatchUnlessCanceled(Exception exception, Cancellatio
 
 #endif
 
-        private static readonly object s_reportedMarker = new();
+        private static readonly object s_reportedMarker = Guid.NewGuid();
```

Comment:
💡 Consider having a comment or test covering the serialization requirement. It seems like it would be trivial to regress.